examples/retry_command.ts

```diff
@@ -0,0 +1,34 @@
+import { Command, CommandStep, Pipeline } from '../src';
+import { YamlSerializer } from '../src/serializers/yaml';
+
+class RetryCommand extends Command {
+    constructor(public retries: number, public c: Command) {
+        super(c.toString());
```

Comment:
```suggestion
        super(c.serialize());
```

as you're interested in the command itself, not the representation for human consumption

Reply:
Are we allowed to do that? since `serialize()` is protected

Reply:
that's what `protected` visibility does - the class and all subclasses